scvi/core/models/base.py

```diff
@@ -179,23 +189,43 @@ def save(self, dir_path: str, overwrite: bool = False):
                     dir_path
                 )
             )
-        torch.save(self.model.state_dict(), os.path.join(dir_path, "model_params.pt"))
-        with open(os.path.join(dir_path, "attr.pkl"), "wb") as f:
+
+        if save_anndata:
+            self.adata.write(
+                os.path.join(dir_path, "adata.h5ad"), **anndata_write_kwargs
+            )
+
+        model_save_path = os.path.join(dir_path, "model_params.pt")
+        attr_save_path = os.path.join(dir_path, "attr.pkl")
+        varnames_save_path = os.path.join(dir_path, "var_names.h5")
+
+        var_names = self.adata.var_names
+        with h5py.File(varnames_save_path, "w") as f:
+            f.create_dataset("var_names", data=var_names.to_numpy().astype("S"))
```

Comment:
why not use https://numpy.org/doc/stable/reference/generated/numpy.save.html? then you don't have to worry about the type